packages/react-cosmos-component-playground/src/components/Tree/index.js

```diff
@@ -0,0 +1,207 @@
+import React from 'react';
+import { arrayOf, shape, number, string, func, bool } from 'prop-types';
+import { FolderIcon, RightArrowIcon, DownArrowIcon } from '../SvgIcon';
+import styles from './index.less';
+import classNames from 'classnames';
+import { match } from 'fuzzaldrin-plus';
+
+const CONTAINER_LEFT_PADDING = 10;
+const INDENT_PADDING = 20;
+
+const FuzzyHighligher = ({ searchText, textToHighlight }) => {
+  if (!searchText) {
+    return <span>{textToHighlight}</span>;
+  }
+
+  const fuzzyMatch = match(textToHighlight, searchText);
+  if (fuzzyMatch.length === 0) {
+    return <span>{textToHighlight}</span>;
+  }
+
+  const highlighted = [];
+  fuzzyMatch.forEach((highlightIndex, index) => {
+    const highlightChar = textToHighlight.slice(
+      highlightIndex,
+      highlightIndex + 1
+    );
+    highlighted.push(
+      <mark key={`highlight-${index}`} className={styles.searchHighlight}>
+        {highlightChar}
+      </mark>
+    );
+    const nextHighlightIndex = fuzzyMatch[index + 1];
+    if (nextHighlightIndex !== highlightIndex + 1) {
+      const unhighlightedChars = textToHighlight.slice(
+        highlightIndex + 1,
+        nextHighlightIndex
+      );
+      highlighted.push(<span key={index}>{unhighlightedChars}</span>);
+    }
+  });
+
+  return <span>{highlighted}</span>;
+};
+
+const TreeFolder = ({
+  node,
+  onSelect,
+  onToggle,
+  selected,
+  nestingLevel,
+  searchText
+}) => {
+  return (
+    <div
+      className={styles.component}
+      onClick={e => {
+        e.preventDefault();
+        e.stopPropagation();
+        onToggle(node, !node.expanded);
+      }}
+    >
+      <div
+        className={styles.componentName}
+        style={{
+          paddingLeft: CONTAINER_LEFT_PADDING + nestingLevel * INDENT_PADDING
```

Comment:
Looks like we need to add `"no-mixed-operators": 0,` to XO rules in package.json, because it's conflicting with Prettier which doesn't like superfluous brackets.

Reply:
Done. Unfortunately there's another Prettier formatting issue. Not sure how you want to deal wit h this (now and going forward).

Reply:
> Unfortunately there's another Prettier formatting issue.

Which one?

> Not sure how you want to deal with this (now and going forward).

Wouldn't want to bother you and this PR with it, but the long term plan is to cherry pick a few ESLint rules remove XO. Too many rules makes for a bad DX and Prettier already takes care of most of the linter's previous responsibilities.

Reply:
OH, I think you meant this commit 78fbe50c0a75d99150bfd536fafad664650606ed

Well, I think adding a file `.prettierignore` with the following fixes it once and for all...

```
package.json
packages/*/package.json
examples/*/package.json
```

Reply:
Did you want this to go in this PR, or will you create a separate one for it?

Reply:
No need. I was already working on something else and put it on that branch.